daemon/containerd/registry_errors.go

```diff
@@ -22,6 +27,21 @@ func translateRegistryError(ctx context.Context, err error) error {
 				log.G(ctx).WithError(derrs).Debug("unable to unmarshal registry error")
 				return fmt.Errorf("%w: %w", cerrdefs.ErrUnknown, err)
 			}
+			if len(derrs) == 0 {
+				if remoteErr.StatusCode == http.StatusUnauthorized || remoteErr.StatusCode == http.StatusForbidden {
+					// Token server may have a unique response
+					var tokenErr tokenServerError
+					if jerr := json.Unmarshal(remoteErr.Body, &tokenErr); jerr == nil && tokenErr.Details != "" {
+						if remoteErr.StatusCode == http.StatusUnauthorized {
+							return cerrdefs.ErrUnauthenticated.WithMessage(fmt.Sprintf("%s - %s", docker.ErrorCodeUnauthorized.Message(), tokenErr.Details))
+						} else {
+							return cerrdefs.ErrPermissionDenied.WithMessage(fmt.Sprintf("%s - %s", docker.ErrorCodeDenied.Message(), tokenErr.Details))
+
+						}
+					}
+
+				}
+			}
```

Comment:
nit; looks like some stray whitespace; the `else` also is redundant, and may start to trip some linters we added / are adding;

```suggestion
			if len(derrs) == 0 {
				if remoteErr.StatusCode == http.StatusUnauthorized || remoteErr.StatusCode == http.StatusForbidden {
					// Token server may have a unique response
					var tokenErr tokenServerError
					if jErr := json.Unmarshal(remoteErr.Body, &tokenErr); jErr == nil && tokenErr.Details != "" {
						if remoteErr.StatusCode == http.StatusUnauthorized {
							return cerrdefs.ErrUnauthenticated.WithMessage(fmt.Sprintf("%s - %s", docker.ErrorCodeUnauthorized.Message(), tokenErr.Details))
						}
						return cerrdefs.ErrPermissionDenied.WithMessage(fmt.Sprintf("%s - %s", docker.ErrorCodeDenied.Message(), tokenErr.Details))
					}
				}
			}
```